src/app/components/AssetDetail/index.jsx

```diff
@@ -0,0 +1,162 @@
+              <small style={{ position: 'relative', top: '-5px' }}><ChangePercent>{change24}</ChangePercent></small>
+              <ArrowIcon
+                style={{ position: 'relative', top: '-5px' }}
+                className={classNames('swapChangeArrow', change24.isZero() ? 'd-none' : null)} 
+                size={.58} dir={change24 < 0 ? 'down' : 'up'} 
+                color={change24 < 0 ? 'danger' : change24 > 0 ? 'success' : null}
+              />
+            </div>
+          </Col>
+          <Col className='col-auto ml-3 mr-3' size='sm'>
+            <div className='py-2'>
+              <div className='text-muted mb-0'>
+                <small>Holdings</small>
+              </div>
+                <Units 
+                  value={assetHoldings || 0} 
+                  symbol={symbol} 
+                  precision={6}
+                />
+            </div>
+          </Col>
+          {marketData.map(({ title, key }, i) => {
+            const flag = (key !== 'availableSupply')
+            return (
+              <Col key={i} className='ml-3 mr-3 d-flex col-auto'>
+                <div className='py-2'>
+                  <div className='text-muted mb-0'>
+                    <small>{title}</small>
+                  </div>
+                  <Units
+                    className='text-nowrap'
+                    value={asset[key]} 
+                    symbol={flag ? '$' : asset.symbol} 
+                    precision={6} 
+                    prefixSymbol={flag}
+                    abbreviate
+                  />
+                </div>
+              </Col>
+            )
+          })}
+          <Col className='d-flex align-items-center flex-row-reverse ml-2 mr-2'>
+            <div className='py-2'>
+              <div className='d-flex flex-nowrap text-muted mb-0'>
+                <Link to={`/swap?to=${symbol}&from=${FROM_ASSET}`}>
+                  <Button className='mr-2' color='success' size='sm' disabled={!receive}>Buy {symbol}</Button>
+                </Link>
+                <Link to={`/swap?to=${TO_ASSET}&from=${symbol}`}>
+                  <Button color='danger' size='sm' disabled={!deposit}>Sell {symbol}</Button>
+                </Link>
+              </div>
+            </div>
+          </Col>
+        </Row>
+      </CardHeader>
+      <CardBody className='text-center'>
+        {isPriceChartLoading ? 
+          <i className='fa fa-spinner fa-pulse'/> :
+          <PriceChart symbol={symbol} chartOpen/> 
+        }
+      </CardBody>
+    </Card>
+    </Layout>
+  )
+}
+
+export default compose(
+  setDisplayName('AssetDetail'),
+  setPropTypes({
+    match: PropTypes.object.isRequired
+  }),
+  withProps((props) => {
+    const symbol = getQuery(props).toUpperCase()
+    return ({
+      symbol
+    })
+  }),
+  connect(createStructuredSelector({
+    asset: (state, { symbol }) => getAsset(state, symbol),
+    assetHoldings: (state, { symbol }) => getHoldingsByAsset(state, symbol),
```

Comment:
This page can be accessed without a connected wallet so I don't think we should show asset holdings in the card you created. Let's keep the purpose focused: to show information about the asset. We can add more specialized cards to this page later, like a card for buy/sell as discussed on slack. Since it'll be designed to only work with connected wallets we can show the holdings there.

Reply:
should i just hide it if holdings is 0? the page works without a wallet connected it just shows 0 for holdings. i think if you do have a wallet connected it would be nice to see and thatd be really quick

Reply:
Sure, that seems reasonable until there's a better place to put it. I'd consider renaming it to "Your holdings" just to be completely clear.